### PR TITLE
fix scalar-tensor issue with CrossEntropyLoss label smoothing, symintify

### PR DESCRIPTION
**Pull request by bdhirsh**
Fixes the fake tensor (and symint) issues with https://github.com/pytorch/pytorch/issues/98571.

In particular, `at::detail::scalar_tensor_static()` seems like a footgun when used in code that needs to be traceable, so we may want to audit its usages more carefully.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98848
* #98031
* #98030
* #98960

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @soumith @desertfire

### CODE REVIEW

File: aten/src/ATen/TensorIndexing.h [hunk 1]
```diff
@@ -369,8 +369,7 @@ static inline Tensor scalarToTensor(
     const TensorOptions& options,
     const at::Device& self_device) {
   if (self_device == at::kCPU) {
-    return at::detail::scalar_tensor_static(
-        v, options.dtype_opt()->toScalarType(), self_device);
+    return at::scalar_tensor(v, options.device(self_device));
```
Comment (ezyang) [inline aten/src/ATen/TensorIndexing.h:372]: seems like you've lost the dtype here

**Comment by ezyang** [comment]
Everything else looks good, but you need to fix ci
